From 955882e3a109a8fd6e1b436c6edcb6a351af8494 Mon Sep 17 00:00:00 2001
From: S-trace <S-trace@list.ru>
Date: Wed, 15 Feb 2017 13:19:38 +0300
Subject: [PATCH] [RUNTIME] Do not crash

Fixes runtime failure:
02-15 13:08:03.886  5701  5701 E AndroidRuntime: FATAL EXCEPTION: main
02-15 13:08:03.886  5701  5701 E AndroidRuntime: Process: org.lineageos.recorder, PID: 5701
02-15 13:08:03.886  5701  5701 E AndroidRuntime: java.lang.NullPointerException
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at java.io.File.<init>(File.java:262)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at org.lineageos.recorder.utils.LastRecordHelper.getOpenIntent(LastRecordHelper.java:54)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at org.lineageos.recorder.sounds.SoundRecorderService.createShareNotification(SoundRecorderService.java:302)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at org.lineageos.recorder.RecorderActivity.toggleSoundRecorder(RecorderActivity.java:289)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at org.lineageos.recorder.sounds.SoundFragment.onClick(SoundFragment.java:81)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at android.view.View.performClick(View.java:5637)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at android.view.View$PerformClick.run(View.java:22433)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:751)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:95)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:154)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6126)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
02-15 13:08:03.886  5701  5701 E AndroidRuntime:        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java

Change-Id: I8487708e6c45ab3aea74ecf4e313269907c2c432
---
 .../java/org/lineageos/recorder/utils/LastRecordHelper.java    | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/app/src/main/java/org/lineageos/recorder/utils/LastRecordHelper.java b/app/src/main/java/org/lineageos/recorder/utils/LastRecordHelper.java
index 12fb17b..3b48468 100644
--- a/app/src/main/java/org/lineageos/recorder/utils/LastRecordHelper.java
+++ b/app/src/main/java/org/lineageos/recorder/utils/LastRecordHelper.java
@@ -7,6 +7,7 @@ import android.content.SharedPreferences;
 import android.net.Uri;
 import android.support.v4.content.FileProvider;
 import android.support.v7.app.AlertDialog;
+import android.util.Log;
 
 import org.lineageos.recorder.R;
 import org.lineageos.recorder.screen.ScreencastService;
@@ -21,6 +22,7 @@ public class LastRecordHelper {
     private static final String KEY_LAST_SOUND_TIME = "sound_last_duration";
     private static final String KEY_LAST_SCREEN_TIME = "screen_last_duration";
     private static final String FILE_PROVIDER = "org.lineageos.recorder.fileprovider";
+    static final String TAG = "LastRecordHelper";
 
     private LastRecordHelper() {
     }
@@ -61,7 +63,13 @@ public class LastRecordHelper {
     }
 
     public static Intent getOpenIntent(Context context, String filePath, String mimeType) {
-        Uri uri = FileProvider.getUriForFile(context, FILE_PROVIDER, new File(filePath));
+        Uri uri = null;
+        try {
+            uri = FileProvider.getUriForFile(context, FILE_PROVIDER, new File(filePath));
+        } catch (NullPointerException error) {
+            Log.e(TAG, "Failed to getUriForFile " + filePath, error);
+            return null;
+        }
         Intent intent = new Intent(Intent.ACTION_VIEW);
         intent.setDataAndType(uri, mimeType);
         intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_GRANT_READ_URI_PERMISSION);
-- 
2.10.2

